### PR TITLE
Observation TreasureMap

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR submits ExecutedObservations to TreasureMap.

### CODE REVIEW

**Review by guynir42 — COMMENTED** [review]
Haven't I already reviewed this yesterday?

File: skyportal/handlers/api/observation.py [hunk 1]
```diff
@@ -874,3 +888,266 @@ def post(self):
...
+            instrument_name=instrument.name,
+            localization_dateobs=localization_dateobs,
+            localization_name=localization_name,
+            localization_cumprob=localization_cumprob,
+            return_statistics=True,
+        )
+
+        observations = data["observations"]
+        num_observations = len(observations)
+        if num_observations == 0:
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/observation.py:1004]: yup.

File: skyportal/handlers/api/observation.py [hunk 2]
```diff
@@ -874,3 +888,266 @@ def post(self):
...
+        if not api_token:
+            raise self.error('Missing allocation information.')
+
+        graceid = event.graceid
+        payload = {"graceid": graceid, "api_token": api_token}
+
+        pointings = []
+        for obs in observations:
+            pointing = {}
+            print(obs)
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/observation.py:1042]: yup.

File: skyportal/handlers/api/observation.py [hunk 3]
```diff
@@ -874,3 +888,266 @@ def post(self):
...
+        payload = {
+            "graceid": graceid,
+            "api_token": api_token,
+            "instrumentid": instrument.treasuremap_id,
+        }
+
+        baseurl = urllib.parse.urljoin(TREASUREMAP_URL, 'api/v0/cancel_all')
+        url = f"{baseurl}?{urllib.parse.urlencode(payload)}"
+        r = requests.post(url=url)
+        r.raise_for_status()
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/observation.py:1152]: yup.

File: skyportal/handlers/api/observation.py [hunk 4]
```diff
@@ -874,3 +888,266 @@ def post(self):
...
+            pointing["time"] = Time(obs["obstime"], format='datetime').isot
+            pointing["depth"] = obs["limmag"]
+            pointing["depth_unit"] = "ab_mag"
+            pointings.append(pointing)
+
+        payload["pointings"] = pointings
+
+        url = urllib.parse.urljoin(TREASUREMAP_URL, 'api/v0/pointings')
+        r = requests.post(url=url, json=payload)
+        r.raise_for_status()
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/observation.py:1054]: yup.

**Review by guynir42 — APPROVED** [review]

**Pull request merged**